students/dave_arasim/session03/list_lab.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+'''List Lab exercise for Session 3, Python 210
+Written by David K. Arasim - 10/17/18'''
+
+print('*'*20, ' Series 1 requirements ', '*'*20)
+print()
+
+fruit_list = ['Apples', 'Pears', 'Oranges', 'Peaches']
+
+print(fruit_list)
+print()
+
+print('Please enter another fruit to add to the list: ', end='')
+new_fruit = input()
+print()
+
+if new_fruit != '':
+    fruit_list.append(new_fruit)
+
+print(fruit_list)
+print()
+
+fruit_list_ct = len(fruit_list)
+
+print('Please enter the fruit number to display (1 - ', fruit_list_ct, '): ', sep='', end='')
+fruit_num = input()
+try:
+    fruit_num = int(fruit_num)
+except ValueError:
+    fruit_num = ''
```

Comment:
Great to see you are already using exception handling! But it looks like this better be combined with the following `if`-`else` block. No need to separate these two out like this. That is,

```
try:
    fruit_num = int(fruit_num)
    if (fruit_num >= 1) and (fruit_num <= fruit_list_ct):
        print(fruit_num, ': ', fruit_list[(fruit_num-1)], sep='')
    else:
        print('That fruit number: ', fruit_num, ' is out of range', sep='')
except ValueError:
    print('Sorry, but that is not a number')
```
Well, actually this approach contains more code in the `try` block then necessary, so it might not be considered good yet. Maybe we should use the Python-specific `else` construct like the following?:
```
try:
    fruit_num = int(fruit_num)
except ValueError:
    print('Sorry, but that is not a number')
else:
    if (fruit_num >= 1) and (fruit_num <= fruit_list_ct):
        print(fruit_num, ': ', fruit_list[(fruit_num-1)], sep='')
    else:
        print('That fruit number: ', fruit_num, ' is out of range', sep='')
```
Frankly, I haven't used `else` on `try`-`except` that much, so I'll need to research more. We'll learn all these next week. Great to see you are looking ahead, David!